tests/saitests/py3/switch.py

```diff
@@ -819,6 +826,14 @@ def sai_thrift_read_port_counters(client, asic_type, port):
     if asic_type == 'mellanox':
         counters_results.append(0)
 
+    # Read SAI_PORT_STAT_IN_DROPPED_PKTS now and insert the cnt at the correct
+    # index in the counter results.
+    if asic_type == 'broadcom':
+        in_drop_pkts_cnt_id = [SAI_PORT_STAT_IN_DROPPED_PKTS]
```

Comment:
if BRCM SAI 11.x, SAI_PORT_STAT_IN_DROPPED_PKTS is not supported, will the call to sai_thrift_get_port_stats  be successful?

Reply:
The counter is supported on Broadcom XGS.  On Broadcom DNX, the SAI call will fail. But this does not fail test today since the  sai thrift call does not check the return value. So on DNX, test will get count value 0.

Reply:
Can we check for broadcom-dnx instead ?

Reply:
Hi @vmittal-msft ,   today the SAI API `sai_thrift_read_port_counters` takes `asic_type` as input, which only tells if the asic if broadcom or not.  If we need to restrict the change to DNX, we have to pass `platform_type` to the API, which requires a large scope of fix because the API is used in many places today.